src/System.Text.Json/tests/JsonEncodedTextTests.cs

```diff
@@ -0,0 +1,325 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Xunit;
+
+namespace System.Text.Json.Tests
+{
+    public static partial class JsonEncodedTextTests
+    {
+        [Fact]
+        public static void Default()
+        {
+            JsonEncodedText text = default;
+            Assert.True(text.EncodedUtf8Bytes.IsEmpty);
+
+            Assert.Equal(0, text.GetHashCode());
+            Assert.Equal("", text.ToString());
+            Assert.True(text.Equals(default));
+
+            JsonEncodedText defaultText = default;
+            object obj = defaultText;
+            Assert.True(text.Equals(obj));
+
+            JsonEncodedText textByteEmpty = JsonEncodedText.Encode(Array.Empty<byte>());
+            Assert.True(textByteEmpty.EncodedUtf8Bytes.IsEmpty);
+            Assert.Equal("", textByteEmpty.ToString());
+
+            JsonEncodedText textCharEmpty = JsonEncodedText.Encode(Array.Empty<char>());
+            Assert.True(textCharEmpty.EncodedUtf8Bytes.IsEmpty);
+            Assert.Equal("", textCharEmpty.ToString());
+
+            Assert.True(textCharEmpty.Equals(textByteEmpty));
+            Assert.Equal(textByteEmpty.GetHashCode(), textCharEmpty.GetHashCode());
+        }
+
+        [Fact]
+        public static void Equals()
+        {
+            string message = "message";
+
+            JsonEncodedText text = JsonEncodedText.Encode(message);
+            JsonEncodedText textCopy = text;
+            JsonEncodedText textDuplicate = JsonEncodedText.Encode(message);
+            JsonEncodedText textDuplicateDiffStringRef = JsonEncodedText.Encode("message");
```

Comment:
This is probably the same string ref, actually (the compiler probably collapsed the literals).  If you want it to be different it needs to be built at runtime.  (string.Create, string.Concat, whatever)